vortex-array/benches/varbinview_compact.rs

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright the Vortex contributors
+
+use divan::Bencher;
+use rand::rngs::StdRng;
+use rand::{Rng, SeedableRng};
+use vortex_array::arrays::VarBinViewArray;
+use vortex_array::builders::VarBinViewBuilder;
+use vortex_array::compute::take;
+use vortex_array::{ArrayRef, IntoArray, ToCanonical};
+use vortex_buffer::Buffer;
+use vortex_dtype::{DType, Nullability};
+use vortex_error::VortexUnwrap;
+
+fn main() {
+    divan::main();
+}
+
+const SMALL_ARGS: &[(usize, usize)] = &[
+    // (output_size, buffer_utilization_pct)
+    (1 << 12, 10),
+    (1 << 12, 90),
+    (1 << 14, 10),
+    (1 << 14, 90),
+];
+
+const LARGE_ARGS: &[(usize, usize)] = &[
+    // (output_size, buffer_utilization_pct)
+    (1 << 19, 10),
+    (1 << 19, 90),
+];
+
+#[divan::bench(args = SMALL_ARGS)]
+fn compact(bencher: Bencher, args: (usize, usize)) {
+    compact_impl(bencher, args);
+}
+
+#[divan::bench(args = LARGE_ARGS, sample_count = 10)]
+fn compact_large(bencher: Bencher, args: (usize, usize)) {
+    compact_impl(bencher, args);
+}
+
+#[divan::bench(args = SMALL_ARGS)]
+fn compact_sliced(bencher: Bencher, args: (usize, usize)) {
+    compact_sliced_impl(bencher, args);
+}
+
+#[divan::bench(args = LARGE_ARGS, sample_count = 10)]
+fn compact_sliced_large(bencher: Bencher, args: (usize, usize)) {
+    compact_sliced_impl(bencher, args);
+}
+
+fn compact_impl(bencher: Bencher, (output_size, utilization_pct): (usize, usize)) {
+    let base_size = (output_size * 100) / utilization_pct;
+    let base_array = build_varbinview_fixture(base_size);
+    let indices = random_indices(output_size, base_size);
+
+    bencher
+        .with_inputs(|| {
+            let taken = take(base_array.as_ref(), &indices).vortex_unwrap();
+            taken.to_varbinview()
+        })
+        .bench_values(|array| array.compact_buffers().vortex_unwrap())
```

Comment:
Do you think there is a way to avoid the memory allocation in the benchmark, it will make the results much more nosiy

Reply:
yea it is not ideal, but I want to measure the overhead of allocations as well, I am experimenting with changes on compact to do less allocations but more cpu, and want to compare that approach with the current one